src/NuGet.Indexing/Extraction/DocumentCreator.cs

```diff
@@ -261,6 +263,56 @@ private static void AddRequiresLicenseAcceptance(Document document, IDictionary<
             }
         }
 
+        private static void AddPackageTypes(Document document, IDictionary<string, string> package)
+        {
+            string value;
+            StringBuilder searchIndex = new StringBuilder();
+            if (package.TryGetValue("flattenedPackageTypes", out value))
+            {
+                if (!string.IsNullOrWhiteSpace(value))
+                {
+                    AddField(document, "FlattenedPackageTypes", value, Field.Index.NOT_ANALYZED);
+
+                    using (var textWriter = new StringWriter())
+                    {
+                        using (var jsonWriter = new JsonTextWriter(textWriter))
+                        {
+                            jsonWriter.WriteStartArray();
+                            foreach (var packageType in value.Split('|'))
+                            {
+                                string[] fields = packageType.Split(':');
+                                if (fields.Length > 0)
+                                {
+                                    jsonWriter.WriteStartObject();
+                                    jsonWriter.WritePropertyName("Type");
+                                    jsonWriter.WriteValue(fields[0]);
+                                    searchIndex.Append(fields[0]);
+                                    searchIndex.Append(" ");
+                                    if (fields.Length > 1)
+                                    {
+                                        jsonWriter.WritePropertyName("Version");
+                                        jsonWriter.WriteValue(fields[1]);
```

Comment:
Might be valuable to put a comment about this there...typically when you see a split like that you assume there's a loop so it's easy to overlook.